modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/AbstractParameterResolverExecutor.java

```diff
@@ -93,11 +101,26 @@ protected ParameterValueResolver parameterValueResolver(ParameterizedElementDecl
         final ParameterModel parameterModel = parameterGroupModel.getParameter(parameterName)
             .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not find parameter with name: '%s' in parameter group: '%s'",
                                                                             parameterName, parameterGroupName)));
-        Object value = extractValue(parameterElement.getValue(),
-                                    artifactHelper.getParameterClass(parameterModel, parameterizedElementDeclaration));
-        if (!parameterModel.getExpressionSupport().equals(NOT_SUPPORTED) && isExpression(value)) {
-          throw new ExpressionNotSupportedException(format("Error resolving value for parameter: '%s' from declaration, it cannot be an EXPRESSION value",
-                                                           parameterName));
+        Class<?> parameterClass = artifactHelper.getParameterClass(parameterModel, parameterizedElementDeclaration);
+        Object value = extractValue(parameterElement.getValue(), parameterClass);
+
+        // TODO: CCNS-26. Improve this.
+        if (value instanceof String && isExpression(value)) {
+
+          if (!parameterModel.getExpressionSupport().equals(NOT_SUPPORTED) && !overrideExpressionSupport) {
+            throw new ExpressionNotSupportedException(format("Error resolving value for parameter: '%s' from declaration, it cannot be an EXPRESSION value",
+                                                             parameterName));
+          }
+
+          value = new TypeSafeExpressionValueResolver<>((String) value,
+                                                        TypedValue.class,
```

Comment:
mmm, I don't think this is correct. I would assume that here it has to go directly the class implementation from the connector's code definition. Even it is weird to see here a TypedValue.class.  Could you explain this change?

Reply:
This is to avoid having to go through the class. Also, keep in mind that this is going to be used as input for resolving another expression. 
Let's say that the class defined in the connector is a String, and that the expression received resolves to a JsonObject. If we resolve this expression directly to a String, when using it as input for the expression that extracts it's fields, that fails. That is becuase DW will have an string as input and is not able to detect it as a Json representation

Reply:
We have discussed this with MA and decided that we could represent input as DW in order to create the value resolver. Same approach could be used in order to resolve either acting parameters for cache or ask DW to generated a minimal representation of a DW script based on the extraction expressions that we would be using for getting acting parameter values.

Reply:
Done in the next commit